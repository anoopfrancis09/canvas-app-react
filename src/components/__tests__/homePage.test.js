import { render, cleanup } from "@testing-library/react";
import React from "react";
import HomePage from "../homePage";
import renderer from "react-test-renderer";
// import "jest-dom/extend-expect";

afterEach(cleanup);
it("renders without  without crashing", () => {
  const div = document.createElement("div");
  render(<HomePage />, div);
});

it("renders the home page", () => {
  const div = document.createElement("div");
  const { getByTestId } = render(<HomePage />, div);

  expect(getByTestId("titleId")).toHaveTextContent("Canvas");
});

it("Check if the image selection button is present", () => {
  const div = document.createElement("div");
  const { getByTestId } = render(<HomePage />, div);

  expect(getByTestId("imageSelectionInputId")).toHaveTextContent(
    "Select an Image file"
  );
});

it("matches snapshot", () => {
  const tree = renderer.create(<HomePage />).toJSON();

  expect(tree).toMatchSnapshot();
});
