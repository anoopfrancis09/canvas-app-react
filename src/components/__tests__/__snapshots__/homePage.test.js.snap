// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <h1
    data-testid="titleId"
  >
    Canvas
  </h1>
  <div
    className="actionContainer"
  >
    <div
      className="actionItemContainer"
      data-testid="imageSelectionInputId"
    >
      <label
        className="custom-file-upload"
      >
        <input
          accept="image/*"
          id="fileSelector"
          onChange={[Function]}
          type="file"
        />
        Select an Image file
      </label>
    </div>
    <div
      className="actionItemContainer"
    >
      <label
        className="custom-file-upload"
      >
        <input
          onChange={[Function]}
          type="file"
        />
        Click to upload image
      </label>
    </div>
  </div>
  <canvas
    className="canvas"
    id="editorCanvas"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseOut={[Function]}
    onMouseUp={[Function]}
  />
</div>
`;
